### DISCUSSION TITLE
Is it possible to allow only certain countries to access the tunnel ?

### DISCUSSION THREAD

**Discussion by lbouriez** [homeassistant-apps/app-cloudflared#209] [Q&A] [answered]
Hello,
Would it be possible to add a security setting to filter the list of countries allowed to access the tunnel ?

**Comment by brenner-tobias** [comment]
Sure, you can use Cloudflare Access for that, following [this article](https://developers.cloudflare.com/cloudflare-one/tutorials/country-rules/).

**Comment by lbouriez** [comment]
Thanks for the link but how can I add a policy to the tunnel that is generated by the addon ?
Or should I add an application in Zero Trust and follow all the article ?
I thought it could have been possible to apply it directly to the tunnel and all the hostname setup through the addon.

**Comment by brenner-tobias** [comment, marked as answer]
You have to add an application in Zero Trust for HomeAssistant and then follow the article.

**Comment by fokcuk** [comment]
I never bothered with setting up in Zero Trust. I just added a FW rule under WAF
